app/scripts/lib/rpc-method-middleware/handlers/request-accounts.js

```diff
@@ -57,13 +58,11 @@ async function requestEthereumAccountsHandler(
     getUnlockPromise,
     hasPermission,
     requestAccountsPermission,
+    openPopup,
   },
 ) {
   if (locks.has(origin)) {
-    res.error = ethErrors.rpc.resourceUnavailable(
```

Comment:
In this situation, there are now two in-flight `eth_requestAccount` requests. What happens with each of them? Would they both succeed? Does the user get prompted twice?

Reply:
yes it returns result twice.

We actually discussed this internally and it seems like there are good cases for both cancelling the other request(s) and to also always return a value

Reply:
Hmm OK. So is the user asked to approve the connection twice?